### PR TITLE
Feature/kinesis lambda dynamodb

### PR DESCRIPTION
**Pull request by rvidis**
Kinesis Lambda DynamoDB streaming test scenario fot Typescript

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: typescript-test-samples/kinesis-lambda-dynamodb/README.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+![System Under Test Description (SUT)](img/system-under-test-description.png)
+
+[Top](#contents)
+---
+
+## About this Example
+
+This example contains an [Amazon Kinesis Data Stream](https://aws.amazon.com/kinesis/data-streams/), [AWS Lambda](https://aws.amazon.com/lambda/) and [Amazon DynamoDB](https://aws.amazon.com/dynamodb/) table core resources.
+
+The Amazon Kinesis Data Stream can stream and data but the AWS Lambda function in this example expects Kinesis Stream Event data to contain a JSON object with 2 properties, `batch` and `id`:
```
Comment (brnkrygs) [inline typescript-test-samples/kinesis-lambda-dynamodb/README.md:54]: Slight typo?

>can stream "and" data but

File: typescript-test-samples/kinesis-lambda-dynamodb/README.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+```json
+{
+    "batch": "string",
+    "id": "string"
+}
+```
+
+ - `batch`: should be unique identifier that represents batch of test records that should all be processed before test is considered completed. Each record in the test batch should have a matching `batch` property value
+ - `id`: unique identifier for each individual record. Each records should have a unique `id` property value
```
Comment (brnkrygs) [inline typescript-test-samples/kinesis-lambda-dynamodb/README.md:64]: Slight typo?

>Each "records" should have

File: typescript-test-samples/kinesis-lambda-dynamodb/metadata.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  "git_repo_url": "https://github.com/aws-samples/serverless-test-samples",
+  "pattern_source": "AWS",
+  "pattern_detail_tabs": [
+      {
+          "title": "Application Code",
+          "filepath": "/src/app.ts"
+      },
+      {
+          "title": "Unit Tests - Mock",
+          "filepath": "/tests/unit/test-handler.test.ts"
```
Comment (brnkrygs) [inline typescript-test-samples/kinesis-lambda-dynamodb/metadata.json:19]: filepath should be: `/src/tests/unit/test-handler.test.ts` I think?

File: typescript-test-samples/kinesis-lambda-dynamodb/metadata.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      },
+      {
+          "title": "Integration Tests",
+          "filepath": "/tests/integration/integ-handler.test.ts"
```
Comment (brnkrygs) [inline typescript-test-samples/kinesis-lambda-dynamodb/metadata.json:23]: filepath should be: `/src/tests/integration/integ-handler.test.ts` I think?

File: typescript-test-samples/kinesis-lambda-dynamodb/src/events/batch-of-five-event.ts
Comment (brnkrygs) [inline typescript-test-samples/kinesis-lambda-dynamodb/src/events/batch-of-five-event.ts:1]: Would prefer this to be a `*.json` file to match other projects.

**Review by brnkrygs — CHANGES_REQUESTED** [review]
Hi @rvidis, thanks for your work on this sample!

The code all worked from what I've been able to tell. I pulled your branch and ran through the README following your instructions and I got passes for all tests. No build errors. There was even a helpful error message when I forgot to set my stack environment variable. Nice job!

I added some inline comments to a couple of files with typos, and found where you might want to update the sample metadata. As soon as those are fixed I think we can merge.

I'd love to see a followup to this that tested some of the Kinesis error handling logic - how partial batch failure behaves, etc. but I don't want to hold this up waiting for that.

**Comment by rvidis** [comment]
Thank you for the review @brnkrygs, I have addressed all comments.

File: typescript-test-samples/kinesis-lambda-dynamodb/metadata.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+          "filepath": "/src/tests/unit/test-handler.test.ts"
+      },
+      {
+          "title": "Integration Tests",
+          "filepath": "/src/tests/integration/integ-handler.test.ts"
+      }
+  ],
+  "authors": [
+      {
+        "name": "Debasis Mitra",
```
Comment (brnkrygs) [inline typescript-test-samples/kinesis-lambda-dynamodb/metadata.json:28]: @rvidis one small tweak yet - wrong name is here.

Want to make sure you get credit! :)

**Review by brnkrygs — CHANGES_REQUESTED** [review]
One small change needed yet: your name on the `metadata.json` file :) then we can GTM

**Review by brnkrygs — APPROVED** [review]
Looks good! Merging.

**Pull request merged**